TeachingRecordSystem/tests/TeachingRecordSystem.Api.Tests/V3/VNext/SetDeceasedTests.cs

```diff
@@ -0,0 +1,111 @@
+        };
+
+        // Act
+        var response = await GetHttpClientWithApiKey().SendAsync(request);
+
+        // Assert
+        Assert.Equal(HttpStatusCode.Forbidden, response.StatusCode);
+    }
+
+    [Fact]
+    public async Task Put_DateOfDeathInFuture_ReturnsErrror()
+    {
+        // Arrange
+        var futureDate = Clock.UtcNow.AddYears(1).ToDateOnlyWithDqtBstFix(isLocalTime: false);
+        var person = await TestData.CreatePerson(p => p.WithTrn());
+        var requestBody = CreateJsonContent(new { DateOfDeath = futureDate });
+        var request = new HttpRequestMessage(HttpMethod.Put, $"v3/persons/deceased/{person.Trn}")
+        {
+            Content = requestBody
+        };
+
+        // Act
+        var response = await GetHttpClientWithApiKey().SendAsync(request);
+
+        // Assert
+        await AssertEx.JsonResponseHasValidationErrorForProperty(
+            response,
+            propertyName: nameof(SetDeceasedCommand.DateOfDeath),
+            expectedError: "Date cannot be in the future.");
+    }
+
+    [Fact]
+    public async Task Put_WithoutDateOfDeath_ReturnsBadRequest()
+    {
+        // Arrange
+        var futureDate = Clock.UtcNow.AddYears(1).ToDateOnlyWithDqtBstFix(isLocalTime: false);
+        var person = await TestData.CreatePerson(p => p.WithTrn());
+        var requestBody = CreateJsonContent(new { DateOfDeath = default(DateOnly?) });
+        var request = new HttpRequestMessage(HttpMethod.Put, $"v3/persons/deceased/{person.Trn}")
+        {
+            Content = requestBody
+        };
+
+        // Act
+        var response = await GetHttpClientWithApiKey().SendAsync(request);
+
+        // Assert
+        Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+    }
+
+    [Fact]
+    public async Task Put_TrnNotFound_ReturnsNotFound()
+    {
+        // Arrange
+        var nonExistentTrn = "1234567";
+        var requestBody = CreateJsonContent(new { DateOfDeath = new DateOnly(1990, 01, 01) });
+        var request = new HttpRequestMessage(HttpMethod.Put, $"v3/persons/deceased/{nonExistentTrn}")
+        {
+            Content = requestBody
+        };
+
+        // Act
+        var response = await GetHttpClientWithApiKey().SendAsync(request);
+
+        // Assert
+        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+    }
+
+    [Fact]
+    public async Task Put_ValidDateOfDeath_ReturnsNoContent()
+    {
+        // Arrange
+        var person = await TestData.CreatePerson(p => p.WithTrn());
+        var requestBody = CreateJsonContent(new { DateOfDeath = new DateOnly(1990, 01, 01) });
+        var request = new HttpRequestMessage(HttpMethod.Put, $"v3/persons/deceased/{person.Trn}")
+        {
+            Content = requestBody
+        };
+
+        // Act
+        var response = await GetHttpClientWithApiKey().SendAsync(request);
+
+        // Assert
+        Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
+    }
+}
```

Comment:
Could use a test that checks a person already marked as deceased can have their date of death changed successfully